### PR TITLE
Update BERT-of-Theseus model card to avoid confusion

### PR DESCRIPTION
**Pull request by JetRunner**
@VictorSanh said this checkpoint cannot be evaluated directly on MNLI. Indeed this model is for intermediate task transfer (in BERT-of-Theseus [arxiv version](https://arxiv.org/abs/2002.02925), it's called a "general-purpose" model) so it doesn't (and shouldn't) contain a classification head. This note helps prevent confusion.

### CODE REVIEW

**Comment by VictorSanh** [comment]
The note makes it clearer! Thanks!

As discussed, I think the confusion arises because of the naming: in the hub, most of the checkpoints with a task in the checkpoint name are full checkpoint with the specific task head. Sometimes there is `finetuned` in the checkpoint name, sometimes not.
This enables a user to directly use a fine-tuned checkpoint for the task at hand out of the box (without any training).

In the long run, it might be worth it to have explicit "guidelines" on checkpoint naming in the hub to avoid these confusions (which should also be mentioned in the model cards if there is any ambiguity). Namely, checkpoints that are general purpose (like `bert-base-uncased`) should be clearly distinguishable from the task-specific checkpoints (like `bert-large-uncased-whole-word-masking-finetuned-squad`). What do you think @julien-c?

**Comment by JetRunner** [comment]
IMO, the most important thing is to allow the authors to specify the example code. A possible design would be: auto generated-example by default but authors can overwrite.
